shared/actions/chat/attachment.js

```diff
@@ -270,89 +329,56 @@ function* onSelectAttachment({payload: {input}}: Constants.SelectAttachment): Ge
     identifyBehavior: yield call(Shared.getPostingIdentifyBehavior, conversationIDKey),
   }
 
-  const channelConfig = Saga.singleFixedChannelConfig([
-    'chat.1.chatUi.chatAttachmentUploadOutboxID',
-    'chat.1.chatUi.chatAttachmentUploadStart',
-    'chat.1.chatUi.chatAttachmentPreviewUploadStart',
-    'chat.1.chatUi.chatAttachmentUploadProgress',
-    'chat.1.chatUi.chatAttachmentUploadDone',
-    'chat.1.chatUi.chatAttachmentPreviewUploadDone',
-    'finished',
-  ])
-
-  const channelMap = (yield call(ChatTypes.localPostFileAttachmentLocalRpcChannelMapOld, channelConfig, {
-    param,
-  }): any)
-  const outboxIDResp = yield Saga.takeFromChannelMap(channelMap, 'chat.1.chatUi.chatAttachmentUploadOutboxID')
-  const {outboxID} = outboxIDResp.params
-  const outboxIDKey = Constants.outboxIDToKey(outboxID)
-  const placeholderMessage = yield call(
-    _appendAttachmentPlaceholder,
-    conversationIDKey,
-    outboxIDKey,
-    preview,
-    title,
-    filename
-  )
-  outboxIDResp.response.result()
-
-  const progressTask = yield fork(function*() {
-    // When we receive the attachment placeholder message from the server, the
-    // local message key basis will change from the outboxID to the messageID.
-    // We need to watch for this so that the uploadProgress gets set on the
-    // right message key.
-    let curKey = placeholderMessage.key
-    while (true) {
-      const {progress, keyChanged, finished} = yield race({
-        progress: Saga.takeFromChannelMap(channelMap, 'chat.1.chatUi.chatAttachmentUploadProgress'),
-        finished: Saga.takeFromChannelMap(channelMap, 'finished'),
-        keyChanged: take(
-          action => action.type === 'chat:outboxMessageBecameReal' && action.payload.oldMessageKey === curKey
-        ),
-      })
-      if (keyChanged) {
-        curKey = keyChanged.payload.newMessageKey
-      } else if (progress) {
-        const {bytesComplete, bytesTotal} = progress.params
-        yield put(Creators.uploadProgress(curKey, bytesComplete / bytesTotal))
-      } else if (finished) {
-        if (finished.error) {
-          yield put(
-            Creators.updateTempMessage(
-              conversationIDKey,
-              {
-                messageState: 'failed',
-                failureDescription: 'upload unsuccessful',
-              },
-              outboxIDKey
-            )
-          )
-        }
-        yield put(Creators.uploadProgress(curKey, null))
-        yield cancel()
-      }
-    }
-  })
+  // TODO This is really hacky, should get reworked
```

Comment:
I would recommend adding back the comment for explanation of what this hackiness does and why it's necessary:

```
// When we receive the attachment placeholder message from the server, the |   |  
// local message key basis will change from the outboxID to the messageID. |   |  
// We need to watch for this so that the uploadProgress gets set on the |   |  
// right message key.
```